maro/simulator/scenarios/finance/common/trader.py

```diff
@@ -0,0 +1,252 @@
+import math
+from enum import Enum
+from collections import OrderedDict
+from maro.simulator.scenarios.finance.common.common import (
+    Action, OrderMode
+)
+
+
+class TradeConstrain(Enum):
+    min_buy_unit = "min_buy_unit"
+    min_sell_unit = "min_sell_unit"
+
+
+class Trader():
+    def __init__(self, trade_constrain: dict):
+        self._order_handlers = OrderedDict()
+        self._slippage_handler = None
+        self._commission_handlers = []
+        self._trade_constrain = OrderedDict()
+        for constrain in trade_constrain:
+            self._trade_constrain[constrain] = trade_constrain[constrain]
+
+        self._trade_number = 0
+
+    @property
+    def supported_orders(self) -> list:
+        return self._order_handlers.keys()
+
+    def order_handler_register(self, order_type: OrderMode, order_handler: callable):
+        self._order_handlers[order_type] = order_handler
+
+    def slippage_handler_register(self, slippage_handler: callable):
+        self._slippage_handler = slippage_handler
+
+    def commission_handler_register(self, commission_handler: callable):
+        self._commission_handlers.append(commission_handler)
+
+    def trade(self, order_action: Action, cur_data: dict, remaining_money: float):
+        # return stock, success, stock_price, number, tax
+        asset = order_action.item_index
+        is_success = False
+        is_trigger = True
+
+        deal_volume = abs(order_action.number)
```

Comment:
add buy/sell to be a field of the Action, don't use +/- to indicate this info

Reply:
done

Reply:
I find there is a ``direction`` in ``Order``, is it used to indicate buy or sell?
What field is used to indicate the quantity?

Reply:
amount is the quantity